source/utilities.tex

```diff
@@ -21135,3 +21135,719 @@
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\returns
+\tcode{frames_.get_allocator()}.
+\end{itemdescr}
+
+\indexlibrarymember{begin}{basic_stacktrace}%
+\indexlibrarymember{cbegin}{basic_stacktrace}%
+\begin{itemdecl}
+const_iterator begin() const noexcept;
+const_iterator cbegin() const noexcept;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\returns
+An iterator referring to the first element in \tcode{frames_}.
+If \tcode{empty()} is \tcode{true},
+then it returns the same value as \tcode{end()}.
+\end{itemdescr}
+
+\indexlibrarymember{end}{basic_stacktrace}%
+\indexlibrarymember{cend}{basic_stacktrace}%
+\begin{itemdecl}
+const_iterator end() const noexcept;
+const_iterator cend() const noexcept;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\returns
+The end iterator.
+\end{itemdescr}
+
+\indexlibrarymember{rbegin}{basic_stacktrace}%
+\indexlibrarymember{crbegin}{basic_stacktrace}%
+\begin{itemdecl}
+const_reverse_iterator rbegin() const noexcept;
+const_reverse_iterator crbegin() const noexcept;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\returns
+\tcode{reverse_iterator(cend())}.
+\end{itemdescr}
+
+\indexlibrarymember{rend}{basic_stacktrace}%
+\indexlibrarymember{crend}{basic_stacktrace}%
+\begin{itemdecl}
+const_reverse_iterator rend() const noexcept;
+const_reverse_iterator crend() const noexcept;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\returns
+\tcode{reverse_iterator(cbegin())}.
+\end{itemdescr}
+
+\indexlibrarymember{empty}{basic_stacktrace}%
+\begin{itemdecl}
+[[nodiscard]] bool empty() const noexcept;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\returns
+\tcode{frames_.empty()}.
+\end{itemdescr}
+
+\indexlibrarymember{size}{basic_stacktrace}%
+\begin{itemdecl}
+size_type size() const noexcept;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\returns
+\tcode{frames_.size()}.
+\end{itemdescr}
+
+\indexlibrarymember{max_size}{basic_stacktrace}%
+\begin{itemdecl}
+size_type max_size() const noexcept;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\returns
+\tcode{frames_.max_size()}.
+\end{itemdescr}
+
+\indexlibrarymember{operator[]}{basic_stacktrace}%
+\begin{itemdecl}
+const_reference operator[](size_type frame_no) const;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\expects
+\tcode{frame_no < size()}.
+
+\pnum
+\returns
+\tcode{frames_[frame_no]}.
+
+\pnum
+\throws
+Nothing.
+\end{itemdescr}
+
+\indexlibrarymember{at}{basic_stacktrace}%
+\begin{itemdecl}
+const_reference at(size_type frame_no) const;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\returns
+\tcode{frames_[frame_no]}.
+
+\pnum
+\throws
+\tcode{out_of_range} if \tcode{frame_no >= size()}.
+\end{itemdescr}
+
+\rSec3[stacktrace.basic.cmp]{Comparisons}
+
+\indexlibrarymember{operator==}{basic_stacktrace}%
+\begin{itemdecl}
+template<class Allocator2>
+friend bool operator==(const basic_stacktrace& x, const basic_stacktrace<Allocator2>& y) noexcept;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\returns
+\tcode{equal(x.begin(), x.end(), y.begin(), y.end())}
```

Comment:
```suggestion
\tcode{equal(x.begin(), x.end(), y.begin(), y.end())}.
```

Reply:
This is an expression. I'm not sure we need a period at the end.

Reply:
See http://wg21.link/support.initlist.access#2, http://wg21.link/support.initlist.range, http://wg21.link/cmp.partialord#5, http://wg21.link/cmp.weakord#2.

Reply:
Agreed. Fixed.